src/lib/logger/js-logger.js

```diff
@@ -5,7 +5,8 @@ import { assign, isError, isString, noop, toArray } from 'lodash'
 import Config from '../../config/config'
 import { addLoggerMonitor } from './monitor'
 
-const connectionErrorRegex = /((connection|network) (error|timeout)|invalid json rpc)/i
+const connectionErrorRegex = /((connection|network) (error|timeout)|invalid json rpc|too many requests)/i
+const rateLimitErrorRegex = /too many requests/i
```

Comment:
for which providers have you verified this is the error returned?

Reply:
Onfinality and I think it was nodies (pokt)